vue-begin/src/components/Employees.vue

```diff
@@ -0,0 +1,187 @@
+        <v-list-tile-action>
+        <v-btn icon @click="showDialog(true, employee)">
+          <v-icon>edit</v-icon>
+        </v-btn>
+      </v-list-tile-action>
+      <v-list-tile-action>
+        <v-btn icon @click="remove(employee.id)">
+          <v-icon>delete</v-icon>
+        </v-btn>
+      </v-list-tile-action>
+      </v-list-tile>
+    </v-list>
+    <div class="text-xs-left">
+      <v-pagination
+        class="pagination-list"
+        v-model="currentPage"
+        :length="Math.floor(employees.length/employeesPerPage)"
+        :total-visible="10"
+      ></v-pagination>
+    </div>
+
+   <v-dialog v-model="dialogShow" persistent max-width="500px">
+    <v-card>
+      <v-card-title>
+        <span class="headline" v-if="dialogForm.dialogId">Update employee</span>
+        <span class="headline" v-else>Add employee</span>
+      </v-card-title>
+      <v-card-text>
+        <v-container grid-list-md>
+          <v-layout wrap>
+            <v-flex xs12 sm6>
+              <v-text-field 
+                label="Employee name"
+                v-model="dialogForm.dialogName" 
+                required/>
+            </v-flex>
+            <v-flex xs12 sm6>
+              <v-text-field
+                label="Salary"
+                type="number"
+                min="0"
+                v-model="dialogForm.dialogSalary" 
+                required>
+              </v-text-field>
+            </v-flex>
+            <v-flex xs12 sm6>
+              <v-text-field 
+                label="Age"
+                type="number"
+                min="0"
+                max="120"
+                v-model="dialogForm.dialogAge" 
+                required>
+              </v-text-field>
+            </v-flex>
+          </v-layout>
+        </v-container>
+      </v-card-text>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn color="blue darken-1" flat
+          @click="showDialog(false)">Close</v-btn>
+        <v-btn color="blue darken-1" flat
+         v-if="dialogForm.dialogId" @click="add()">Add</v-btn>
+         <v-btn color="blue darken-1" flat
+         v-else @click="update(dialogForm.dialogId)">Update</v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+  </div>
+</template>
+
+<script>
+  import * as APIService from '../APIService';
+
+const createDefaultItem = () => ({
+  dialogName: '',
+  dialogSalary: '',
+  dialogShow: false,
+  dialogId: true,
+  dialogAge: null,
+})
+
+  export default {
+    name: 'Employees',
+    props: {
+      msg: String
+    },
+    data() {
+        return {
+            employees: [],
+            employeesPerPage: 20,
+            currentPage: 1,
+            dialogShow: false,
+            dialogForm:{
+              dialogId: null,
+              dialogName: null,
+              dialogSalary: null,
+              dialogAge: null,
+            }
+        };
+    },
+
+    computed: {
+      paginatedEmployees() {
+        return this.employees.slice(
```

Comment:
You can set start and end indexes in seperate variables for better readability.

Reply:
ok